osdep/terminal-unix.c

```diff
@@ -437,10 +422,19 @@ static MP_THREAD_VOID terminal_thread(void *ptr)
         if (fds[1].revents & POLLIN) {
             int8_t c = -1;
             (void)read(stop_cont_pipe[0], &c, 1);
-            if (c == PIPE_STOP)
+            if (c == PIPE_STOP) {
                 do_deactivate_getch2();
-            else if (c == PIPE_CONT)
+                if (isatty(tty_out)) {
+                    (void)write(tty_out, TERM_ESC_RESTORE_CURSOR,
+                                sizeof(TERM_ESC_RESTORE_CURSOR) - 1);
+                }
+                // make do with SIGSTOP since it's effectively the same and
+                // there's no way to get a function pointer to the "real
+                // default" handler.
+                raise(SIGSTOP);
```

Comment:
There's a possible race here where if we've received a continue signal and it's queued in the pipe, we'll go to stop state before processing the cont signal. No clue how to fix it though.

But unlikely to happen in practice, and if it does happen, just sending another cont signal should fix it. So maybe not worth worrying about it (?).